enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthTestBase.java

```diff
@@ -385,11 +417,33 @@ protected void assertKeyIsMap( ResourceIterator<Map<String, Object>> r, String k
             return Stream.of( new CountResult( nNodes ) );
         }
 
+        @Procedure( name = "test.allowedProcedure1", allowed = "role1", mode = Procedure.Mode.READ )
+        public Stream<AuthProcedures.StringResult> allowedProcedure1()
+        {
+            Result execute = db.execute( "RETURN 'foo' AS foo" );
```

Comment:
This query does not need READ access to the database (we currently do not have an EXECUTE permission, so you are currently allowed to use Cypher as a calculator with the custom role). If we change the query to use MATCH the test will actually prove something.